docs/source/callbacks.rst

```diff
@@ -94,6 +94,8 @@ Lightning has a few built-in callbacks.
     :nosignatures:
     :template: classtemplate.rst
 
+    BackboneLambdaFinetuningCallback
+    BaseFinetuningCallback
```

Comment:
Should `BaseFinetuningCallback` be included in the documentation?
The documentation on this class very sparse. For example it should atleast state in class docstring that the user needs to implement the `finetunning_function` and `freeze_before_training` method to get a working subclass.

Reply:
As `BaseFinetuningCallback` is currently part of public API, I'd include the class in the docs, or we could make the class private if it's not supposed to be exposed to users. What do you think?

Reply:
If it is part of the public API then we should probably include the class in the docs (else we need to deprecate it 2 releases from now). IMO we should probably improve the documentation of this class.

Reply:
Yes, I was planning to spend some time on the doc and provide a better name to `BackboneLambdaFinetuningCallback ` which is too long.
Thanks @akihironitta to worked on this.

Reply:
@tchaton sounds good to me :]